WebPhapp/backend/server.js

```diff
@@ -961,15 +1054,83 @@ app.get('/api/v1/dispensers/prescriptions/open/:dispenserID', (req, res) => {
     }
 });
 
+/*
+About:
+    An api endpoint that returns a single dispenser given a dispenserID.
+Examples:
+    Directly in terminal:
+        >>> curl "http://localhost:5000/api/v1/dispensers/single/1"
+    To be used in Axois call:
+        .get("/api/v1/dispensers/single/1")
+Returns:
+    {
+        dispenserID (int),
+        name (string),
+        location (string),
+        phone (int)
+    }
+*/
+app.get('/api/v1/dispensers/single/:dispenserID', (req,res) => { // auth.checkAuth([Role.Government]),
+    var dispenserID = parseInt(req.params.dispenserID);
+
+    var dispensers = readJsonFileSync(
+        __dirname + '/' + "dummy_data/dispensers.json").dispensers;
+
+    dispensers = dispensers.filter(function(elem) {
+        return dispenserID === elem.dispenserID;
+    });
+
+    if(dispensers.length > 1) {
+        console.log('/api/v1/dispensers/single/: error: too many dispenserID matches');
+        res.status(400).send(false);
+    }
+    else if(dispensers.length === 0) {
+        console.log('/api/v1/dispensers/single/: error: no dispenserID match');
+        res.status(400).send(false);
+    }
+    else {
+        dispenser = dispensers[0]
+        console.log('/api/v1/dispensers/single/: returning dispenser with ID ' + dispenserID.toString());
+        res.status(200).send(dispenser);
+    }
+});
+
+/*
+About:
+    An api endpoint that returns all dispensers.
+Examples:
+    Directly in terminal:
+        >>> curl "http://localhost:5000/api/v1/dispensers/all"
+    To be used in Axois call:
+        .get("/api/v1/dispensers/all")
+Returns:
+    [
+        {
+            prescriberID (int),
```

Comment:
nit: need to update comments in name queries to include first and last name fields :)

Reply:
dispenser info includes name, dispenserID, location, and phone. Name is not first and last. Let me know if we are not on the same page there.

Reply:
you right